src/analysis/analyzer.ts

```diff
@@ -61,6 +64,10 @@ export class Analyzer extends AnalyzerGen implements vs.Disposable {
 		this.serverSetSubscriptions({
 			subscriptions: ["STATUS"]
 		});
+
+		// Hook error subscriptions so we can try and get diagnostic info if this happens.
+		this.registerForServerError(e => this.requestDiagnosticsUpdate());
```

Comment:
Yeah, I was trying to come up with a good way - this seemed reasonable, and we could also ping every few mins if they tend to not come through after a crash to try and get some "recent" data instead (though worth seeing what the data is like first).

There's been a *lot* of crashes/errors in SDK 1.20, I'm hoping they're mostly all related to the issue I raised that was fixed and 1.21 will be much better!